test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_fill.cpp

```diff
@@ -0,0 +1,164 @@
+
+        error = clFinish(queue);
+        test_error(error, "clFinish failed");
+
+        std::vector<cl_char> output_data(data_size);
+        error = clEnqueueReadImage(queue, image, CL_TRUE, origin, region, 0, 0, output_data.data(), 0, nullptr, nullptr);
+
+        for (size_t i = 0; i < data_size; i++)
+        {
+            CHECK_VERIFICATION_ERROR(pattern, output_data[i], i);
+        }
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        image = create_image_2d(context, CL_MEM_READ_WRITE, &formats, img_width,
+                          img_height, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        return CL_SUCCESS;
+    }
+
+    
+    int        img_width = 512;
+    int        img_height = 512;
+    const size_t data_size = img_width * img_height * 4;
+    const size_t origin[3]={ 0, 0, 0}, region[3]={ img_width, img_height, 1 };
+    const char pattern = 0x10;
+    const cl_uint fill_color[4] = { pattern, pattern, pattern, pattern };
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+
+    clMemWrapper image;
+    
+};
+
+struct FillBufferKHR : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandFillBufferKHR(
+                command_buffer, nullptr, buffer, &pattern, sizeof(cl_char), 0,
+                data_size, 0, nullptr, nullptr, nullptr);
+        
+        test_error(error, "clCommandFillBufferKHR failed");
+        
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+        
+        error = clFinish(queue);
+
+        test_error(error, "clFinish failed");
+        
+        std::vector<cl_char> output_data(data_size);
+        error = clEnqueueReadBuffer(queue, buffer, CL_TRUE, 0,
+                                    data_size, output_data.data(), 0, nullptr,
+                                    nullptr);
+        test_error(error, "clEnqueueReadBuffer failed");
+
+        for (size_t i = 0; i < data_size; i++)
+        {
+            CHECK_VERIFICATION_ERROR(pattern, output_data[i], i);
+        }
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        buffer = clCreateBuffer(context, CL_MEM_READ_WRITE, data_size, nullptr, &error);
+        test_error(error, "Unable to create buffer");            
```

Comment:
No need for new buffer I think, you could use one of those created in `BasicCommandBufferTest::SetUp`

Reply:
Done